bcda/encryption/encryption.go

```diff
@@ -0,0 +1,95 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"github.com/CMSgov/bcda-app/bcda/database"
+	"github.com/CMSgov/bcda-app/bcda/models"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+// NewEncryptionKey generates a random 256-bit key for Encrypt() and
+// Decrypt(). It panics if the source of randomness fails.
+func newEncryptionKey() *[32]byte {
+	key := [32]byte{}
+	_, err := io.ReadFull(rand.Reader, key[:])
+	if err != nil {
+		panic(err)
+	}
+	return &key
+}
+
+// Encrypt encrypts data using 256-bit AES-GCM.  This both hides the content of
+// the data and provides a check that it hasn't been altered. Output takes the
+// form nonce|ciphertext|tag where '|' indicates concatenation.
+func encrypt(plaintext []byte, key *[32]byte) (ciphertext []byte, err error) {
```

Comment:
And sorry, upon even just the tiniest bit of reflection, that was worded rudely. The concern remains: that this stuff is easy to mess up. But the second sentence there was unnecessary, and jerkish of me.

Reply:
This is cribbed from https://github.com/gtank/cryptopasta. This repo is specifically there to prevent people form rolling their own crypto.  I just wanted to obscure those mid level methods with some things to make our lives easier.

Reply:
Apologies for the probably dumb question, but what gives you confidence that the cryptopasta reference is a solid one? Is gtank well-known/well-trusted in the Go/cryptography community? (FWIW, I see FiloSottile in some of the issue comments, and he's definitely well-known in the cryptography community.)

Aside from the "authority" question, I'm more concerned that this hasn't been touched at all in two years: no issue or commit or PR activity since 2016. Is there perhaps a more actively maintained set of "Go crypto" recommendations?

My starting point for these things is usually this article: <https://latacora.micro.blog/2018/04/03/cryptographic-right-answers.html>.

Reply:
I read that whole page several times.  Crypto pasta seems to be a core go dev.  His pinned tweet references working with FiloSottile https://twitter.com/gtank__